### DISCUSSION TITLE
asp core and oracle efcore with elsa issue

### DISCUSSION THREAD

**Discussion by adnanelhabashy** [elsa-workflows/elsa-core#4046] [Design & tech]
iam trying to start a project using elsa with Oracle entity framework core 

below is the code for my program.cs file 
i hace done all the step 

using Elsa;
using Elsa.Persistence.EntityFramework.Core.Extensions;
using Elsa.Persistence.EntityFramework.Oracle;

using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.
builder.Services.AddControllersWithViews();

var elsaSection = builder.Configuration.GetSection("Elsa");
var ConnectionString = builder.Configuration.GetConnectionString("OracleConnection");

builder.Services
                .AddElsa(elsa => elsa
                    .UseEntityFrameworkPersistence(ef => ef.UseOracle(ConnectionString ,b=> b.MigrationsAssembly("ElsaWorkFlow")))
                    .AddConsoleActivities()
                    .AddHttpActivities(elsaSection.GetSection("Server").Bind)
                    .AddQuartzTemporalActivities()
                    .AddWorkflowsFrom<Program>()
                );

// Elsa API endpoints.
builder.Services.AddElsaApiEndpoints();

// For Dashboard.
builder.Services.AddRazorPages();

var app = builder.Build();

// Configure the HTTP request pipeline.
if (!app.Environment.IsDevelopment())
{
    app.UseExceptionHandler("/Home/Error");
    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
    app.UseHsts();
}

app.UseHttpsRedirection();
app.UseStaticFiles();

app.UseRouting();
app.UseHttpActivities();
app.UseAuthorization();

app.MapControllerRoute(
    name: "default",
    pattern: "{controller=Home}/{action=Index}/{id?}");

app.UseEndpoints(endpoints =>
{
    // Elsa API Endpoints are implemented as regular ASP.NET Core API controllers.
    endpoints.MapControllers();

    // For Dashboard.
    endpoints.MapFallbackToPage("/_Host");
});

app.Run();

and below is my appsettings.json file 

{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft.AspNetCore": "Warning"
    }
  },
  "Elsa": {
    "Server": {
      "BaseUrl": "https://localhost:5001"
    }
  },

  "ConnectionStrings": {

    "OracleConnection": "Data Source=(DESCRIPTION = (ADDRESS = (PROTOCOL = TCP)(HOST = localhost)(PORT = 1521))(CONNECT_DATA = (SERVER = DEDICATED) (FAILOVER_MODE = (TYPE = select,session) (METHOD = preconnect)) (SID = XE)));User ID=adnan;Password=123456"
  },
  "AllowedHosts": "*"
}

and i keep getting the below exception 

Oracle.ManagedDataAccess.Client.OracleException: 'ORA-01435: user does not exist
ORA-06512: at line 8'

it says the user doesnot exist 
even though i tried the connection string with normal oracle efcore setup without else is working fine 

hope i could find help here as iam getting frustrated

**Comment by mohdali** [comment]
I might be wrong, but I think it is complaining about the schema name, not the user id. I think if you create a schema with the name Elsa then it will be able to create the tables.

Alternatively, does the user have full acccess to create tables and schemas? Is it a SYSDBA?

**Reply by adnanelhabashy** [reply]
i have another database user with the name elsa and i did try it but same error shows up and yes the user has all privileges to create tables like sysdba

**Reply by mohdali** [reply]
Can you add `DBA Privilege=SYSDBA;` to your connection string and try?
